server/testnet.go

```diff
@@ -0,0 +1,240 @@
+				if err != nil {
+					_ = os.RemoveAll(outputDir)
+					return err
+				}
+
+				config.Moniker = nodeDirName
+				gaiaConfig := gaiacfg.DefaultConfig()
+				gaiaConfig.Name = nodeDirName
+				gaiaConfig.CliRoot = clientDir
+				gaiaConfig.GenTx.Overwrite = true
+
+				// Run `init gen-tx` and generate initial transactions
+				cliPrint, genTxFile, err := gentxWithConfig(config, gaiaConfig, ctx, cdc, appInit)
+				if err != nil {
+					return err
+				}
+
+				// Save private key seed words
+				name := fmt.Sprintf("%v.json", "key_seed")
+				writePath := filepath.Join(clientDir)
+				file := filepath.Join(writePath, name)
+				err = cmn.EnsureDir(writePath, 0700)
+				if err != nil {
+					return err
+				}
+				err = cmn.WriteFile(file, cliPrint, 0600)
+				if err != nil {
+					return err
+				}
+
+				// Gather gentxs folder
+				name = fmt.Sprintf("%v.json", nodeDirName)
+				writePath = filepath.Join(gentxsDir)
+				file = filepath.Join(writePath, name)
+				err = cmn.EnsureDir(writePath, 0700)
+				if err != nil {
+					return err
+				}
+				err = cmn.WriteFile(file, genTxFile, 0644)
+				if err != nil {
+					return err
+				}
+			}
+
+			// Generate genesis.json
+			for i := 0; i < nValidators; i++ {
+
+				nodeDirName := cmn.Fmt("%s%d", nodeDirPrefix, i)
+				nodeDir := filepath.Join(outputDir, nodeDirName, "gaiad")
+				clientDir := filepath.Join(outputDir, nodeDirName, "gaiacli")
+				gentxsDir := filepath.Join(outputDir, "gentxs")
+				gaiaConfig := gaiacfg.DefaultConfig()
+				gaiaConfig.Name = nodeDirName
+				gaiaConfig.CliRoot = clientDir
+				gaiaConfig.Init.ChainID = "something-something"
+				gaiaConfig.Init.GenTxs = true
+				gaiaConfig.Init.Overwrite = true
+				gaiaConfig.Init.GenTxsDir = gentxsDir
+				config.SetRoot(nodeDir)
+
+				// Run `init` and generate genesis.json
+				_, _, _, err := initWithConfig(config,gaiaConfig, ctx,cdc,appInit)
+				if err != nil {
+					return err
+				}
+
+				pvFile := filepath.Join(nodeDir, config.BaseConfig.PrivValidator)
+				pv := pvm.LoadFilePV(pvFile)
+				genVals[i] = types.GenesisValidator{
+					PubKey: pv.GetPubKey(),
+					Power:  1,
+					Name:   nodeDirName,
+				}
+			}
+
+			for i := 0; i < nNonValidators; i++ {
+				nodeDirName := cmn.Fmt("%s%d", nodeDirPrefix, i+nValidators)
+				nodeDir := filepath.Join(outputDir, nodeDirName, "gaiad")
+				config.SetRoot(nodeDir)
+
+				err := os.MkdirAll(filepath.Join(nodeDir, "config"), nodeDirPerm)
+				if err != nil {
+					_ = os.RemoveAll(outputDir)
+					return err
+				}
+
+				config.Moniker = nodeDirName
+				//TODO: Fix it!v
+				initWithConfig(config,nil,ctx,cdc,appInit)
+			}
+
+			if populatePersistentPeers {
+				err := populatePersistentPeersInConfigAndWriteIt(config)
+				if err != nil {
+					_ = os.RemoveAll(outputDir)
+					return err
+				}
+			}
+
+			fmt.Printf("Successfully initialized %v node directories\n", nValidators+nNonValidators)
+			return nil
+			},
+	}
+	cmd.Flags().IntVar(&nValidators, "v", 4,
```

Comment:
let's stick with the convention in the rest of the sdk for now and just use `viper.GetString` instead of cobra "var" flags